pkg/stripe/analytics_telemetry.go

```diff
@@ -0,0 +1,169 @@
+package stripe
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/google/go-querystring/query"
+	"github.com/google/uuid"
+
+	"github.com/spf13/cobra"
+
+	"github.com/stripe/stripe-cli/pkg/useragent"
+	"github.com/stripe/stripe-cli/pkg/version"
+)
+
+//
+// Public types
+//
+
+// TelemetryContextKey is the key for the telemetry context
+type TelemetryContextKey struct{}
+
+// TelemetryClientKey is the key for the telemetry client
+type TelemetryClientKey struct{}
```

Comment:
Not a blocker, but it's a best practice to keep these keys unexported, and instead define functions that allow you to get or set the the value. In particular, this lets callers avoid making type assertions, as you've had to do in a few places. For example:

```
func WithEventMetadata(context.Context, *CLIAnalyticsEventMetadata) context.Context
func GetEventMetadata(context.Context) *CLIAnalyticsEventMetadata
```

Reply:
I like this much better, thanks!